src/meetChat/hooks/useChatScroll.tsx

```diff
@@ -0,0 +1,111 @@
+import { useState, useEffect, RefObject } from "react";
+import { chatType } from "../types/chatTypes.ts";
+
+interface ChatScrollProps {
+  messages: chatType[];
+  isLoading: boolean;
+  mainRef: RefObject<HTMLDivElement>;
+}
+
+/**
+ * 채팅 스크롤 관리를 위한 커스텀 훅
+ * 새 메시지, 이전 메시지 로드 시 스크롤 위치를 적절히 처리합니다.
+ */
+export function useChatScroll({
+  messages,
+  isLoading,
+  mainRef,
+}: ChatScrollProps) {
+  const [prevScrollHeight, setPrevScrollHeight] = useState(0);
+  const [initialLoad, setInitialLoad] = useState(true);
+  const [isScrolling, setIsScrolling] = useState(false);
+  const [prevMessagesLength, setPrevMessagesLength] = useState(0);
+  const [initialScrollApplied, setInitialScrollApplied] = useState(false);
+
+  // 최하단으로 스크롤하는 함수 - 애니메이션 없음
+  const scrollToBottomImmediate = () => {
+    if (mainRef.current) {
+      // 애니메이션 없이 즉시 스크롤
+      mainRef.current.scrollTop = mainRef.current.scrollHeight;
```

Comment:
```suggestion
        mainRef.current.scrollTo({
        top: mainRef.current.scrollHeight,
        behavior: "smooth",
      });
```

위 코드를 사용하면 부드럽게 하단으로 스크롤을 내릴 수 있어요!